File: app/src/main/java/com/icerockdev/babenko/ui/splash/SplashPresenter.java
package com.icerockdev.babenko.ui.splash;

import com.arellomobile.mvp.InjectViewState;
import com.icerockdev.babenko.managers.SharedPreferencesManager;
import com.icerockdev.babenko.ui.base.BasePresenter;

@InjectViewState
public class SplashPresenter extends BasePresenter<SplashView> {
    private final SharedPreferencesManager manager;

    public SplashPresenter(SharedPreferencesManager manager) {
        this.manager = manager;
    }

    public void animationFinished() {
        if (manager.isUserLogged()) {
            getViewState().startImagesActivity();
        } else {
            getViewState().startLoginActivity();
        }
    }
}

File: app/src/main/java/com/icerockdev/babenko/ui/splash/SplashPresenter.kt
package com.icerockdev.babenko.ui.splash

import com.arellomobile.mvp.InjectViewState
import com.icerockdev.babenko.managers.SharedPreferencesManager
import com.icerockdev.babenko.ui.base.BasePresenter

@InjectViewState
class SplashPresenter(private val manager: SharedPreferencesManager) : BasePresenter<SplashView>() {

    fun animationFinished() {
        if (manager.isUserLogged) {
            viewState.startImagesActivity()
        } else {
            viewState.startLoginActivity()
        }
    }
}
